gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala

```diff
@@ -88,12 +89,29 @@ trait BasicScanExecTransformer extends LeafTransformSupport with BaseDataSource
   }
 
   override protected def doValidateInternal(): ValidationResult = {
+    def addStringFields(
```

Comment:
nit: addStringFields -> appendStringFields, and maybe extract this function out to generate all the fields needed for validation.